### PR TITLE
Add performance tests with its make target

### PR DESCRIPTION
**Pull request by palasanu**
Add performance tests with its make target
Add /version endpoint

## Make command example

```
❯ make test-performance API_TOKEN=$token
Building docker image 'performance:latest'...
sha256:4d52cfda7fcb62d1115afab6a9c8a20b529d9342e2265bc9ea04e1d168eb4ecf
Done!

Running benchmark on http://cluster-registry-api:8080/readyz for 10s ...
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Benchmark ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
 Total Requests: 13284 ┃ Requests/s: 1328 ┃ Average: 0.038 ┃ 99.9th: 0.294
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

Running benchmark on http://cluster-registry-api:8080/livez for 10s ...
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Benchmark ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
 Total Requests: 324371 ┃ Requests/s: 32433 ┃ Average: 0.002 ┃ 99.9th: 0.011
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

Running benchmark on http://cluster-registry-api:8080/api/v1/clusters for 10s ...
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Benchmark ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
 Total Requests: 18656 ┃ Requests/s: 1865 ┃ Average: 0.027 ┃ 99.9th: 0.830
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

Geting cluster name... Done! Got cluster01-prod-useast1.
Running benchmark on http://cluster-registry-api:8080/api/v1/clusters/cluster01-prod-useast1 for 10s ...
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Benchmark ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
 Total Requests: 31720 ┃ Requests/s: 3172 ┃ Average: 0.016 ┃ 99.9th: 0.047
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
```
More info in the docs update.

## Motivation and Context

We needed some performance tests so we can see how the project performs over time.

After this, a mechanism that runs these tests automatically should be added.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by victorvarza — CHANGES_REQUESTED** [review]
Looking good! 
Small changes from my side as performance test tool should be considered a separate app and it's makefile targets build/release should live outside of the main Makefile.

File: Makefile [hunk 1]
```diff
@@ -145,7 +145,19 @@ test-e2e:
 	@. local/.env.local && go test -race github.com/adobe/cluster-registry/test/e2e -count=1 -v
 	$(shell pwd)/local/cleanup.sh
 
+IMAGE_TAG_PERFORMENCE=latest
```
[the remark this answers is not in the document]
Comment (palasanu) [reply Makefile:148]: Sound good, I will make the changes when I find some time

**Comment by palasanu** [comment]
run tests

File: Makefile [hunk 2]
```diff
@@ -15,11 +15,11 @@ VERSION?=$(shell cat VERSION | tr -d " \t\n\r")
 # The ldflags for the go build process to set the version related data.
 GO_BUILD_LDFLAGS=\
 	-s \
-	-X $(GO_PKG)/version.Revision=$(BUILD_REVISION)  \
```
Comment (palasanu) [inline Makefile:18]: I tried to pass the variables to the config package, but it did not work. I tried on the database package and it worked, I'm could not figure out why it did not work on the config package so I left them on the main.

**Review by victorvarza — APPROVED** [review]

**Pull request merged**